core/src/main/java/org/apache/iceberg/ManifestGroup.java

```diff
@@ -104,6 +127,29 @@ ManifestGroup caseSensitive(boolean newCaseSensitive) {
   }
 
   /**
+   * Returns a iterable of scan tasks. It is safe to add entries of this iterable
+   * to a collection as {@link DataFile} in each {@link FileScanTask} is defensively
+   * copied.
+   * @return a {@link CloseableIterable} of {@link FileScanTask}
+   */
+  public CloseableIterable<FileScanTask> scanTasks() {
+    Iterable<CloseableIterable<FileScanTask>> tasks = entries((manifest, entries) -> {
+      PartitionSpec spec = specsById.get(manifest.partitionSpecId());
+      String schemaString = SchemaParser.toJson(spec.schema());
+      String specString = PartitionSpecParser.toJson(spec);
+      ResidualEvaluator residuals = ResidualEvaluator.of(spec, dataFilter, caseSensitive);
+      return CloseableIterable.transform(entries, e -> new BaseFileScanTask(
+          e.copy().file(), schemaString, specString, residuals));
+    });
+
+    if (PLAN_SCANS_WITH_WORKER_POOL && manifests.size() > 1) {
+      return new ParallelIterable<>(tasks, ThreadPools.getWorkerPool());
```

Comment:
I think a better way to configure this is to leave `PLAN_SCANS_WITH_WORkER_POOL` in the data scan and add `ManifestGroup.planWith(ExecutorService)`. If the service is set, then it will use that service, otherwise it uses the single-threaded option. That way, this class is more generic and doesn't depend on the environment.

Reply:
+1. Nice idea!